hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +187,75 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
     long committed = nodeStat.getCommitted().get();
     return new DatanodeStorageReport(capacity, used, remaining, committed);
   }
+
+  /**
+   * This GET API provides the information of all datanodes for which decommissioning is initiated.
+   * @return the wrapped  Response output
+   */
+  @GET
+  @Path("/decommission/info")
+  public Response getDatanodesDecommissionInfo() {
+    // Command to execute
+    List<String> commandArgs = Arrays.asList("ozone", "admin", "datanode", "status", "decommission", "--json");
+
+    return getDecommissionStatusResponse(commandArgs);
```

Comment:
since recon run in its user space, and this needs ozone in path to be invoked. If Path is not set properly, this can fail. check if user have capability to set env variable, path or conf if required.

Reply:
Since `src/shell/ozone/ozone` is being distributed as part of ozone binaries, this should allow user to set explicitly if path is not set correctly. If it’s installed from Apache opensource on-prem, we have to put the ozone script in PATH manually, the classpath construction will be handled by the ozone script and It should work automatically if Java Home is set.